### PR TITLE
getPosts

### PR DESCRIPTION
**Pull request by itsmebipul00**
[VIEW](https://github.com/itsmebipul00/nextxt_frontend/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=dotfile&show-deleted-files=true&show-viewed-files=true&w=1)

Many commented files are todos or reference for future

### CODE REVIEW

**Review by archit7448 — APPROVED** [review]
good work looks good to merge

File: src/Components/AllPosts.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { StyledPost } from './styles/Post.styled'
+
+import { useSelector, useDispatch } from 'react-redux'
+
+import { useEffect } from 'react'
+
+import { getAllPosts } from '../Features/postsSlice'
+
+import moment from 'moment'
```
Comment (iRohitGaur) [inline src/Components/AllPosts.js]: `Moment.js is a legacy project, now in maintenance mode. In most cases, you should choose a different library.`
- [Moment JS NPM](https://www.npmjs.com/package/moment)
- [Blog](https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore)

File: src/Components/Header.js [hunk 2]
```diff
@@ -23,13 +40,55 @@ function Header() {
 		dispatch(toggleTheme(theme))
 		setTheme(theme)
 	}
+	const [collapseNav, setCollapseNav] = useState(true)
+
+	//dont show nav on auth page
+	//show nava for small scrreens
```
Comment (iRohitGaur) [inline src/Components/Header.js:46]: suggestion: remove commented code

File: src/Components/NewPost.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+		newPostData.append('postImgs', newPost.file)
+
+		dispatch(createNewPost(newPostData))
+
+		setNewPost(initialState)
+
+		setModal(prev => !prev)
+
+		window.location.reload()
```
Comment (iRohitGaur) [inline src/Components/NewPost.js]: query: why are you reloading here?

File: src/Components/NewPost.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+								<input
+									id='gif'
+									type='file'
+									accept='image/gif'
+									onChange={handleMedia}
+									disabled={!!postGif ? true : false}
+									hidden
+								/>
+								<FluentGif16Regular />
+							</label> */}
```
Comment (iRohitGaur) [inline src/Components/NewPost.js:172]: suggestion: remove commented code

File: src/Components/NewPost.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+								</video>
+							))}
+
+						<div className='call-to-actions'>
+							<label htmlFor='img-vid'>
+								<input
+									id='img-vid'
+									type='file'
+									accept='image/jpeg,image/png,image/webp,video/mp4,video/quicktime,video/webm'
+									onChange={uploadNewFile}
```
Comment (iRohitGaur) [inline src/Components/NewPost.js:157]: suggestion: check if you can do `image/*, video/*` here

File: src/Services/postsServices.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	// 		const res = await axios.post(
+	// 			`${API}/api/posts/draft/${userInfo._id}`,
+	// 			newPost,
+	// 			config
+	// 		)
+	// 		return res.data
+	// 	} catch (error) {
+	// 		throw error
+	// 	}
+	// },
```
Comment (iRohitGaur) [inline src/Services/postsServices.js:36]: suggestion: remove commented code

**Review by iRohitGaur — APPROVED** [review]
Nice work! UI looks good.

Gave some suggestions, please look into it.

Good to merge with updates 👍🏻

File: src/Components/AllPosts.js (hunk 1 above)
Comment (itsmebipul00) [reply src/Components/AllPosts.js]: Thankyou, didnt know this. Changed to date fns, works fine.

File: src/Components/NewPost.js (hunk 3 above)
Comment (itsmebipul00) [reply src/Components/NewPost.js]: Thanks for ponting this . This was  to readily update state and show them in view. I have updated this and used useEffect and props which works without reaload

File: src/Components/NewPost.js (hunk 4 above)
Comment (itsmebipul00) [reply src/Components/NewPost.js:172]: for future reference

File: src/Components/NewPost.js (hunk 5 above)
Comment (itsmebipul00) [reply src/Components/NewPost.js:157]: yes thankyou updated it

File: src/Services/postsServices.js (hunk 6 above)
Comment (itsmebipul00) [reply src/Services/postsServices.js:36]: future reference

**Pull request merged**